### ISSUE TITLE
Find a way to build our custom Dolphin build on Mac, Linux and Android(?) and include them into the pack

### ISSUE THREAD

**Issue by MoArtis** [MoArtis/ResidentEvilSeamlessHdProject#83]
It seems that some people are playing the pack on mac and android.

Let's try to give them the same experience as on PC (4:3, no music bug...).

**Comment by Soapsuds** [comment]
Hey! Is the project using some kind of build bot? If not I would be super happy to walk through how I did the appimage version on Linux. I don't have like any experience using a build bot but I bet we could figure something out :)

**Comment by MoArtis** [comment]
Thanks for proposing that. We are not using a build bot since it's likely that the project will not need anymore changes on Dolphin. :) That being said, I'm not against a build bot if it allows us to get clean linux and mac builds that can be easily downloaded by the community.

**Comment by Hunk4TH** [comment]
Can anyone help me get this built on Manjaro Linux?

**Comment by MoArtis** [comment]
@Hunk4TH I cannot help you sadly but at least make sure to use that repo for your attempt at building Dolphin for linux:
https://github.com/MoArtis/dolphin
Good luck! 👍

**Comment by Hunk4TH** [comment]
> @Hunk4TH I cannot help you sadly but at least make sure to use that repo for your attempt at building Dolphin for linux:
> https://github.com/MoArtis/dolphin
> Good luck! +1

Thanks man and thanks for all your guys hard work! :)

**Comment by Soapsuds** [comment]
Hey so Project slippi's dolphin fork got appimages working recently, I'm pretty sure I can copy some stuff from them and get a some kind of build working! Just letting you guys know I'll start looking into it again :)

@Hunk4TH AFAIK it's going to be really hard to get it to build on a recent distro, but I'll make sure to update this when I get something working. If you want to get your hands dirty I was able to build on Ubuntu 16.04 and then package it as an appimage and that kinda works

**Comment by Hunk4TH** [comment]
> 
> 
> Hey so Project slippi's dolphin fork got appimages working recently, I'm pretty sure I can copy some stuff from them and get a some kind of build working! Just letting you guys know I'll start looking into it again :)
> 
> @Hunk4TH AFAIK it's going to be really hard to get it to build on a recent distro, but I'll make sure to update this when I get something working. If you want to get your hands dirty I was able to build on Ubuntu 16.04 and then package it as an appimage and that kinda works
Sounds good! I was thinking of an appimage as as well. I think that would probably be a better solution. What "kinda works"?

**Comment by Soapsuds** [comment]
Okay, so after a lot of fiddling I've got a build working! 
If you want to check it out go to this url
https://github.com/Soapsuds/dolphin/actions/runs/190517191 
and download the linux artifact.

A few things:
I need to learn how to squash commits for the PR, this should have been like 2 commits XD
Vulkan doesn't work on this build, not sure what that's about yet
After you download it the appimage will use your global dolphin directory, to have it not do that you need to make a directory with the same name as the appimage + .home eg "Dolphin_Emulator-adeb0ff-x86_64.AppImage.home" from there you can load all your textures and things without having it mess with a regular dolphin install.

**Comment by MoArtis** [comment]
@Soapsuds Nice! Great job!

I have no idea how the portable mode is supposed to work on Mac but on PC we just need to add a "portable.txt" file next to the executable.

![image](https://user-images.githubusercontent.com/3904610/89097581-ba667c80-d412-11ea-95ef-ee1d1256635c.png)

**Comment by Soapsuds** [comment]
Hey, thanks :) 

Portable.txt does work the same, but because it's packed into an appimage the directory next to the executable isn't writable :/ 
It looks like slippi is still using global paths eg .config on Linux but changed the directory name to SlippiOnline. I'm thinking we could either do something similar or instruct users to make the .appimage.home directory (we could even package it as a zip with the appimage and the directory included). 

I'm thinking that .appimage.home is probably good enough, but I'm curious to know what you think :)

**Comment by MoArtis** [comment]
Sadly I very rarely had to use a mac so I have no idea what you are referring to. :D

What we are usually aim for is providing a frictionless as possible process for the end user. Unzip / Install and it should "just work".  

One other thing we need to think about are the config files. We pre-configured the emulator and you might need to grab these files included in our final packs.

**Comment by Soapsuds** [comment]
Oh no worries, the stuff I was referring to is actually just for Linux XD😅 
I feel good about the way I'm packing it up. Getting it started should be as easy as extract and run, but it would be really nice to have another Linux tester. Hopefully once the PR is open we can find someone to try it out.

The configuration files do get packed up correctly, so that should be no problem.

The only blocker I feel is that vulkan doesn't work. I'd want to either find a way to get it working, or disable it.

**Comment by MoArtis** [comment]
I'm guessing that the Vulkan issue is not coming from your build process though. 🤔 Is Vulkan working on the official build (especially from the same version)?

**Comment by MoArtis** [comment]
Sorry for the confusion, re-reading the thread again and I'm now clear that we are talking about a Linux build. 😅

**Comment by Hunk4TH** [comment]
@Soapsuds you still working on this? I've been so busy lately haven't had a chance to check out the appimage yet. Sorry for the late response.

**Comment by Soapsuds** [comment]
Nope! It's done :) link is in the update links discord or right here! https://mega.nz/file/rUpTzD6Z#7pnWRmwEVgh6cg6bf8-nn1vqvuCnyOnu7sXBJpoJ0sI

Dm me on discord if you notice anything weird with it!

**Comment by Hunk4TH** [comment]
> Nope! It's done :) link is in the update links discord or right here! https://mega.nz/file/rUpTzD6Z#7pnWRmwEVgh6cg6bf8-nn1vqvuCnyOnu7sXBJpoJ0sI
> 
> Dm me on discord if you notice anything weird with it!

Wow thanks! I haven't used Discord in forever so I appreciate you posting it here! Will do if I find anything :)
